tiledb/sm/query/writer.cc

```diff
@@ -1007,7 +950,7 @@ Status Writer::compute_write_cell_ranges(
 
   auto domain = array_schema_->domain();
   auto dim_num = array_schema_->dim_num();
-  auto subarray = (const T*)subarray_;
+  auto subarray = (const T*)&subarray_flat_[0];
```

Comment:
Do we need a check here for `!subarray_flat_.empty()`?

Reply:
No, but we'd better add an assertion here. The subarray *must* be set at this point and, hence, the flat subarray vec as well.